pilot/pkg/proxy/envoy/watcher_test.go

```diff
@@ -122,27 +130,55 @@ func TestWatchCerts(t *testing.T) {
 		called <- true
 	}
 
+	// test modify file event
 	ctx, cancel := context.WithCancel(context.Background())
+	go watchCerts(ctx, []string{tmpFile.Name()}, watchFileEvents, 50*time.Millisecond, callbackFunc)
 
-	go watchCerts(ctx, []string{name}, watchFileEvents, 50*time.Millisecond, callbackFunc)
+	// sleep one second to make sure the watcher is set up before change is made
+	time.Sleep(time.Second)
+
+	// modify file
+	_, err = tmpFile.Write([]byte("foo"))
+	if err != nil {
+		t.Errorf("failed to update file %s: %v", tmpFile.Name(), err)
+	}
+	err = tmpFile.Sync()
```

Comment:
nit: in golang, it's more idiomatic when just returning an error to check the result in one line:

``golang
if err = tmpFile.Sync; err != nil {
    t.Errorf("...")
}
```